src/aurelia-hot-module-reload.ts

```diff
@@ -224,13 +224,16 @@ export class HmrContext {
       return; // first load
     }
 
-    const cssPluginModuleId = this.loader.applyPluginToUrl(moduleId, 'css-resource-plugin');
+    const extensionIndex = moduleId.lastIndexOf('.');
+    const moduleExtension = moduleId.substring(extensionIndex + 1);
+    const pluginName = `${moduleExtension}-resource-plugin`;
+    const cssPluginModuleId = this.loader.applyPluginToUrl(moduleId, pluginName);
```

Comment:
If such a plugin doesn't exist, it would be good to throw a meaningful error to the user, otherwise it might be confusing.

Rest should be fine.

Reply:
@niieani, there is no way to ask the loader if it is 'aware' of a plugin. I will make a change to check if the `analyzedModule` as specified by `cssPluginModuleId` is undefined and report an error for that so at least it's not a `Cannot read property 'resources' of undefined` error

Reply:
ok

Reply:
@niieani Is this blocked on anything? I can't tell if the naming convention would for for stylus .styl files or not.

Reply:
@kayle just waiting for extra check from @AdamWillden and we can merge this as a workaround for the moment.

Reply:
@niieani @kayle sorry for the delay. Note the additional friendly error check in the latest fixup commit:

```diff
+    if (typeof analyzedModule === 'undefined') {
+      console.error(`Unable to find module, check the plugin exists and the module has been loaded with the expected plugin`);
+      return;
+    } else 
```

Reply:
@EisenbergEffect would we want to use `aurelia-logger` here instead of `console`?

Reply:
@niieani IIRC none of the errors use `aurelia-logger` - I copied how the existing errors are handled.

I would argue that is out of scope of this PR and should be tackled in it's own right.